strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -309,50 +309,41 @@ def fromscovmat(self, V, modes=None):
             if n > self.nlen:
                 raise ValueError("Covariance matrix is larger than the number of subsystems.")
 
-        # convert to xp ordering
-        rotmat = changebasis(n)
-        VV = np.dot(np.dot(np.transpose(rotmat), V), rotmat)
-
-        A = VV[0:n, 0:n]
-        B = VV[0:n, n : 2 * n]
-        C = VV[n : 2 * n, n : 2 * n]
-        Bt = np.transpose(B)
-
-        if n < self.nlen:
-            # reset modes to be prepared back to the vacuum state
-            for mode in modes:
-                self.loss(0.0, mode)
-
-        rows = modes.reshape(-1, 1)
-        cols = modes.reshape(1, -1)
-        self.nmat[rows, cols] = 0.25 * (A + C + 1j * (B - Bt) - 2 * np.identity(n))
-        self.mmat[rows, cols] = 0.25 * (A - C + 1j * (B + Bt))
+        mode_ind = np.sort(np.append(2 * np.array(modes), 2 * np.array(modes) + 1))
+        # Delete current entries of covariance and any correlations to other modes
+        self.covs[:, mode_ind, :] = 0
+        self.covs[:, :, mode_ind] = 0
+        # Set new covariance elements
+        self.covs[np.ix_(np.arange(self.covs.shape[0], dtype=int), mode_ind, mode_ind)] = V
 
     def qmat(self, modes=None):
```

Comment:
likely this method can be deleted since it is relevant only the Gaussian backend.